src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -31,31 +31,31 @@ class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
 
     const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
 
-    const ids: string[] = this.splitIdsList(args.options.ids);
+    const ids: string[] = Utils.splitAndTrim(args.options.ids);
     const idsChunks: string[][] = [];
 
     while (ids.length) {
       idsChunks.push(ids.splice(0, 20));
     }
 
-    idsChunks.forEach(async (idsChunk: string[], index: number) => {
-      const requestOptions: any = {
-        url: requestUrl,
-        headers: {
-          'accept': 'application/json;odata=nometadata'
-        },
-        responseType: 'json',
-        data: {
-          ids: idsChunk
-        }
-      };
-
-      await request.post(requestOptions);
-
-      if(index === idsChunks.length - 1) {
-        cb();
-      }
-    });
+    Promise.all(
+      idsChunks.map(async (idsChunk: string[]) => {
+        const requestOptions: any = {
+          url: requestUrl,
+          headers: {
+            'accept': 'application/json;odata=nometadata',
+            'content-type': 'application/json'
+          },
+          responseType: 'json',
+          data: {
+            ids: idsChunk
+          }
+        };
+
+        await request.post(requestOptions);
```

Comment:
If we use `Promise.all`, I don't think we need to `await` here and we should use `return request.post(...)` which returns a Promise. Once it's resolved, `Promise.all` will proceed.

Reply:
yes of course.. sorry for that. I think I was doing it between some other stuff and just moved the code and forgot to remove the await